### PR TITLE
backoff to snowflake connection establishment

### PR DESCRIPTION
**Pull request by pivnicek**
rel: https://github.com/keboola/connection/issues/576

### CODE REVIEW

**Comment by Halama** [comment]
@pivnicek there were proble in tests in master. I've fixed it and rebased this PR.

**Comment by pivnicek** [comment]
@Halama I saw that, thank you!

File: src/Snowflake/Connection.php [hunk 1]
```diff
@@ -46,6 +46,7 @@ public function __construct(array $options)
 
         $port = isset($options['port']) ? (int)$options['port'] : 443;
         $tracing = isset($options['tracing']) ? (int)$options['tracing'] : 0;
+        $maxBackoffAttempts = isset($options['maxBackoffAttempts']) ? (int)$options['maxBackoffAttempts'] : 5;
```
Comment (Halama) [inline src/Snowflake/Connection.php]: there should be also space after type cast. Unfortunately it is not catched by code style checker.

**Comment by Halama** [comment]
@pivnicek can you please check these issues?

**Comment by pivnicek** [comment]
yes, sorry,  mmt.

File: src/Snowflake/Connection.php (hunk 1 above)
Comment (pivnicek) [inline src/Snowflake/Connection.php]: image

yes, it wasn't checked by default

**Comment by Halama** [comment]
there are still CS issue in build...

**Comment by pivnicek** [comment]
gah, I really should check the build outputs!

**Comment by Halama** [comment]
This is that message with translation:

image

```
Nov 14 08:46:40 kbc-vpc-sapi-WORKER-i-908dbb09 Zend_Log:  {
message: Queue process error
priority: WARN
app: keboola-connection
pid: 8520
cliCommand: ./scripts/cli.php --ansi storage:queue:receive main_fast 2
module: storage
messageId: 523805a3-f0f6-4dc3-aafe-d7c9bd298015
newMessageId: 666ccc91-d9d4-4896-9da7-c61de52fc165
retryCount: 1
delaySecs: 120
jobId: 18995608
queueMessage: {
id: 523805a3-f0f6-4dc3-aafe-d7c9bd298015
body: {
jobId: 18995608
timestamp: 2016-11-14T08:46:39+01:00
retryCount: 1
}
}
exception: {
message: Initializing Snowflake connection failed: odbc_connect(): SQL error: [unixODBC][Snowflake][Snowflake] (4) 
      REST request for URL https://keboola.snowflakecomputing.com:443/session/v1/login-request?databaseName="sapi_258"&schemaName=&warehouse="KEBOOLA_PROD"&requestId=e6823fd9-5e9b-4c74-a8a5-513b999bc6c3 failed: Unknown error (http error).
    , SQL state S1000 in SQLConnect
code: 0
attachment: https://connection.keboola.com/admin/utils/logs?file=2016/11/14/08/2016-11-14-08-46-39-58296bdfe48cb-exception
}
component: JobExecutor
}
```

**Comment by pivnicek** [comment]
should probably retry that too eh

**Comment by Halama** [comment]
I think we can check that `S1000` it is present in both versions.

**Comment by Halama** [comment]
These logs without translated message are from older worker with ODBC drivers `ODBC 2.12.86
54` the second one newer where the messages are translated has `ODBC 2.12.87`.

I'll shut down the older one.

**Comment by pivnicek** [comment]
ok.  even if that `S1000` is a generic error, worst thing that happens is it tries to reconnect 5 times.

**Comment by Halama** [comment]
yes. it is ok. I have tried some common errors like bad login and it is returning other codes so it should be ok.

**Comment by Halama** [comment]
:)
```
ODBC 2.12.89	
SNOW-22102: Fixed a potential deadlock when the main application thread waiting for a result chunk, being downloaded by an asynchronous thread, times out.
SNOW-21795, SNOW-24366, SNOW-24589, SNOW-24519: Improved handling of connection failures and re-establishing a connection.
```

**Pull request merged**